source/client-side-encryption/client-side-encryption.rst

```diff
@@ -1256,24 +1278,39 @@ One of the strings:
 - "AEAD_AES_256_CBC_HMAC_SHA_512-Random"
 - "Indexed"
 - "Unindexed"
+- "RangePreview"
 
-The result of explicit encryption with the "Indexed" algorithm must be processed by the server to insert or query. Drivers MUST document the following behavior:
+The result of explicit encryption with the "Indexed" or "RangePreview" algorithm must be processed by the server to insert or query. Drivers MUST document the following behavior:
 
-   To insert or query with an "Indexed" encrypted payload, use a ``MongoClient`` configured with ``AutoEncryptionOpts``.
+   To insert or query with an "Indexed" or "RangePreview" encrypted payload, use a ``MongoClient`` configured with ``AutoEncryptionOpts``.
    ``AutoEncryptionOpts.bypassQueryAnalysis`` may be true. ``AutoEncryptionOpts.bypassAutoEncryption`` must be false.
 
+NOTE: The Range algorithm is experimental only. It is not intended for public use.
+
 contentionFactor
 ^^^^^^^^^^^^^^^^
-contentionFactor only applies when algorithm is "Indexed".
+contentionFactor only applies when algorithm is "Indexed" or "RangePreview".
 It is an error to set contentionFactor when algorithm is not "Indexed".
 
+NOTE: The Range algorithm is experimental only. It is not intended for public use.
+
 queryType
 ^^^^^^^^^
 One of the strings:
 - "equality"
+- "rangePreview"
+
+queryType only applies when algorithm is "Indexed" or "RangePreview".
+It is an error to set queryType when algorithm is not "Indexed" or "RangePreview".
```

Comment:
As I recall, this error should be thrown on libmongocrypt level, not in the driver?

Reply:
my mistake, we have a check for this in the driver too

Reply:
libmongocrypt returns an error if queryType is [if algorithm is not "Indexed" or "RangePreview"](https://github.com/mongodb/libmongocrypt/blob/a8c6a9717214b5c8531116a9a6550c697b8cd043/src/mongocrypt-ctx-encrypt.c#L2419-L2423) or if queryType [does not match the index type](https://github.com/mongodb/libmongocrypt/blob/a8c6a9717214b5c8531116a9a6550c697b8cd043/src/mongocrypt-ctx-encrypt.c#L2469-L2496).